### PR TITLE
Fix NPE during swap-restore plan generation

### PR DESCRIPTION
**Pull request by ggezer**
Related to issues seen in https://github.com/Alluxio/alluxio/issues/10429

### CODE REVIEW

File: core/server/worker/src/main/java/alluxio/worker/block/management/tier/SwapRestoreTask.java [hunk 1]
```diff
@@ -141,12 +141,15 @@ public BlockManagementTaskResult run() {
         long blockId = tierIterator.next();
         try {
           BlockMeta nextBlockFromTier = mEvictorView.getBlockMeta(blockId);
-          moveOutBytes -= nextBlockFromTier.getBlockSize();
-          if (lastTier) {
-            blocksToRemove.add(nextBlockFromTier.getBlockId());
-          } else {
-            blocksToTransfer.add(BlockTransferInfo.createMove(nextBlockFromTier.getBlockLocation(),
-                nextBlockFromTier.getBlockId(), destLocation));
+          if (nextBlockFromTier != null) {
```
Comment (apc999) [inline core/server/worker/src/main/java/alluxio/worker/block/management/tier/SwapRestoreTask.java:144]: can you check if line 189 too? May that create NPE too?

Comment (jiacheliu3) [inline core/server/worker/src/main/java/alluxio/worker/block/management/tier/SwapRestoreTask.java:144]: Could you log a debug message when this is null? That way we can know if the `if` block is executed. It always frustrates me when some logic should have been executed but somehow is not (or even worse, I don't know if it is or not).

**Review by jiacheliu3 — APPROVED** [review]
LGTM with a minor comment, thanks!

File: core/server/worker/src/main/java/alluxio/worker/block/management/tier/SwapRestoreTask.java (hunk 1 above)
Comment (ggezer) [reply core/server/worker/src/main/java/alluxio/worker/block/management/tier/SwapRestoreTask.java:144]: Right. Fixed that too.

**Comment by ggezer** [comment]
alluxio-bot, merge this please

**Comment by ggezer** [comment]
alluxio-bot, cherry-pick this to branch-2.6 please

**Comment by ggezer** [comment]
alluxio-bot, cherry-pick this to branch-2.5 please

**Comment by ggezer** [comment]
alluxio-bot, cherry-pick this to branch-2.4 please

**Pull request merged**